sky/utils/kubernetes_utils.py

```diff
@@ -578,3 +621,322 @@ def from_resources(cls,
+            session. It is either 'NODEPORT' or 'PORTFORWARD'
+        namespace: Kubernetes namespace to use
+        port_fwd_proxy_cmd_path: str; path to the script used as Proxycommand
+            with 'kubectl port-forward'
+        port_fwd_proxy_cmd_template: str; template used to create
+            'kubectl port-forward' Proxycommand
+    """
+    # Fetch IP to connect to for the jump svc
+    ssh_jump_ip = get_external_ip(network_mode)
+    if network_mode == KubernetesNetworkingMode.NODEPORT:
+        ssh_jump_port = get_port(ssh_jump_name, namespace)
+        ssh_jump_proxy_command = construct_ssh_jump_command(
+            private_key_path, ssh_jump_port, ssh_jump_ip)
+    # Setting kubectl port-forward/socat to establish ssh session using
+    # ClusterIP service to disallow any ports opened
+    else:
+        ssh_jump_port = LOCAL_PORT_FOR_PORT_FORWARD
+        vars_to_fill = {
+            'ssh_jump_name': ssh_jump_name,
+            'local_port': ssh_jump_port,
+        }
+        backend_utils.fill_template(port_fwd_proxy_cmd_template,
+                                    vars_to_fill,
+                                    output_path=port_fwd_proxy_cmd_path)
+        ssh_jump_proxy_command = construct_ssh_jump_command(
+            private_key_path, ssh_jump_port, ssh_jump_ip,
+            port_fwd_proxy_cmd_path)
+    return ssh_jump_proxy_command
+
+
+def setup_sshjump_svc(ssh_jump_name: str, namespace: str,
+                      service_type: KubernetesServiceType):
+    """Sets up Kubernetes service resource to access for SSH jump pod.
+
+    This method acts as a necessary complement to be run along with
+    setup_sshjump_pod(...) method. This service ensures the pod is accessible.
+
+    Args:
+        sshjump_name: Name to use for the SSH jump service
+        namespace: Namespace to create the SSH jump service in
+        service_type: Networking configuration on either to use NodePort
+            or ClusterIP service to ssh in
+    """
+    # Fill in template - ssh_key_secret and sshjump_image are not required for
+    # the service spec, so we pass in empty strs.
+    content = fill_sshjump_template('', '', ssh_jump_name, service_type.value)
+    # Create service
+    try:
+        kubernetes.core_api().create_namespaced_service(namespace,
+                                                        content['service_spec'])
+    except kubernetes.api_exception() as e:
+        # SSH Jump Pod service already exists.
+        if e.status == 409:
+            ssh_jump_service = kubernetes.core_api().read_namespaced_service(
+                name=ssh_jump_name, namespace=namespace)
+            curr_svc_type = ssh_jump_service.spec.type
+            if service_type.value == curr_svc_type:
+                # If the currently existing SSH Jump service's type is identical
+                # to user's configuration for networking mode
+                logger.warning(
```

Comment:
Should we use `debug` instead. It seems the user do not need to know the underlying implementation details.

Reply:
Agreed! Changed now.